src/Avalonia.Controls/TextBoxTextInputMethodClient.cs

```diff
@@ -0,0 +1,41 @@
+using System;
+using Avalonia.Controls.Presenters;
+using Avalonia.Input.TextInput;
+using Avalonia.VisualTree;
+
+namespace Avalonia.Controls
+{
+    internal class TextBoxTextInputMethodClient : ITextInputMethodClient
+    {
+        private TextPresenter _presenter;
+        private IDisposable _subscription;
+        public Rect CursorRectangle => _presenter?.GetCursorRectangle() ?? default;
```

Comment:
Why do we need a Rectangle here instead of just the position?

Reply:
Both ibus and fcitx want an actual rectangle

Reply:
Actually, iOS might ask for an cursor rectangle for an arbitrary text position

Reply:
I guess we can always pass the caret bounding box. The caret itself is not always a rectangle.

Reply:
I think bounding box would be expected, yes.